File: data/readable_format/fortran/file1277.f90
PROGRAM TestSLA_GBRPVGRW
  IMPLICIT NONE
  REAL :: result
  INTEGER, PARAMETER :: n = 3, kl = 1, ku = 1, ncols = 3, ldab = 3, ldafb = 3
  REAL, DIMENSION(ldab, ncols) :: AB = RESHAPE( &
       [1.0, 2.0, 0.0, &
        3.0, 4.0, 1.0, &
        0.0, 3.0, 5.0], &
       [ldab, ncols])
  REAL, DIMENSION(ldafb, ncols) :: AFB = RESHAPE( &
       [0.5, 1.0, 0.0, &
        1.5, 2.0, 0.5, &
        0.0, 1.5, 2.5], &
       [ldafb, ncols])

  result = SLA_GBRPVGRW(n, kl, ku, ncols, AB, ldab, AFB, ldafb)
  PRINT *, "Result: ", result

CONTAINS

  REAL FUNCTION SLA_GBRPVGRW( N, KL, KU, NCOLS, AB, LDAB, AFB, LDAFB )
    IMPLICIT NONE
    INTEGER, INTENT(IN) :: N, KL, KU, NCOLS, LDAB, LDAFB
    REAL, INTENT(IN) :: AB( LDAB, * ), AFB( LDAFB, * )
    INTEGER :: I, J, KD
    REAL :: AMAX, UMAX, RPVGRW
    RPVGRW = 1.0
    KD = KU + 1
    DO J = 1, NCOLS
       AMAX = 0.0
       UMAX = 0.0
       DO I = MAX( J-KU, 1 ), MIN( J+KL, N )
          AMAX = MAX( ABS( AB( KD+I-J, J)), AMAX )
       END DO
       DO I = MAX( J-KU, 1 ), J
          UMAX = MAX( ABS( AFB( KD+I-J, J ) ), UMAX )
       END DO
       IF ( UMAX /= 0.0 ) THEN
          RPVGRW = MIN( AMAX / UMAX, RPVGRW )
       END IF
    END DO
    SLA_GBRPVGRW = RPVGRW
  END FUNCTION SLA_GBRPVGRW

END PROGRAM TestSLA_GBRPVGRW

File: data/readable_format/cpp/file1277.cpp
#include <iostream>
#include <vector>
#include <algorithm> // For std::max and std::min
#include <cmath>     // For std::abs

float sla_gbrpvgrw(int n, int kl, int ku, int ncols, float* ab, int ldab, float* afb, int ldafb) {
    int i, j, kd;
    float amax, umax, rpvgrw = 1.0f;
    kd = ku;

    for (j = 0; j < ncols; ++j) {
        amax = 0.0f;
        umax = 0.0f;
        for (i = std::max(j - ku, 0); i < std::min(j + kl + 1, n); ++i) {
            amax = std::max(std::abs(ab[(kd + i - j) + j * ldab]), amax);
        }
        for (i = std::max(j - ku, 0); i <= j; ++i) {
            umax = std::max(std::abs(afb[(kd + i - j) + j * ldafb]), umax);
        }
        if (umax != 0.0f) {
            rpvgrw = std::min(amax / umax, rpvgrw);
        }
    }

    return rpvgrw;
}

int main() {
    const int n = 3, kl = 1, ku = 1, ncols = 3, ldab = 3, ldafb = 3;
    std::vector<float> AB = {
        1.0f, 2.0f, 0.0f,
        3.0f, 4.0f, 1.0f,
        0.0f, 3.0f, 5.0f
    };
    std::vector<float> AFB = {
        0.5f, 1.0f, 0.0f,
        1.5f, 2.0f, 0.5f,
        0.0f, 1.5f, 2.5f
    };

    float result = sla_gbrpvgrw(n, kl, ku, ncols, AB.data(), ldab, AFB.data(), ldafb);
    std::cout << "Result: " << result << std::endl;
    return 0;
}